feat: 6017 - from proof page, show previously added prices

monsieurtanuki opened the pull request:
### What
- When we added prices to an existing proof, we couldn't see which prices were already added.
- In this PR, we load the related existing prices and we display them in the "add price" page.

### Screenshot
![Screenshot_1740314439](https://github.com/user-attachments/assets/343410fd-a3a1-4738-80b6-a5bd11b69802)

### Fixes bug(s)
- Closes: #6017

### Files
New files:
* `price_existing_amount_card.dart`: Card that displays an existing amount.
* `price_existing_amount_field.dart`: Text field that displays a read-only amount for an existing price.

Impacted files:
* `price_add_product_card.dart`: checking if the barcode to be added already existed in the already added prices too, not only among the current amounts.
* `price_model.dart`: now maintaining a list of already added prices
* `price_proof_page.dart`: now we load async'ly the proof-related prices in order to display them in the next "add prices" page
* `product_price_add_page.dart`: now displaying the previously added prices

monsieurtanuki committed:
feat: 6017 - from proof page, show previously added prices

New files:
* `price_existing_amount_card.dart`: Card that displays an existing amount.
* `price_existing_amount_field.dart`: Text field that displays a read-only amount for an existing price.

Impacted files:
* `price_add_product_card.dart`: checking if the barcode to be added already existed in the already added prices too, not only among the current amounts.
* `price_model.dart`: now maintaining a list of already added prices
* `price_proof_page.dart`: now we load async'ly the proof-related prices in order to display them in the next "add prices" page
* `product_price_add_page.dart`: now displaying the previously added prices

teolemon commented:
The title being "Amount" is weird. It might be the Barcode potentially ? But there might be some cases where it's the category. 
What happens in those cases by the way ? Have you already added prices on products in bulk ?

monsieurtanuki commented:
> The title being "Amount" is weird.

It's the same subtitle as for "new prices".
What about something like "Price already added"? (You'll note the "history" icon btw)

> It might be the Barcode potentially ? But there might be some cases where it's the category. 

Anyway the product is already described (image, title, barcode)

> What happens in those cases by the way ?

Probably not very informative I'm afraid. I did hesitate between this issue and the "add price for category" issue.
That said, I can work (a little) on how to display "category prices" as read-only for this issue.

> Have you already added prices on products in bulk ?

Erm, no, but I cannot see the connection with the current PR. Here we display the prices already added to a single proof.

monsieurtanuki committed:
After first comments

Impacted files:
* `price_existing_amount_card.dart`: now displays the category, if relevant
* `price_existing_amount_field.dart`: now displays the "PricePer" component, if relevant

monsieurtanuki commented:
@teolemon I've just
* added the category
* added the "price per" component
* changed the subtitle to "Price previously added"

![Screenshot_1740385785](https://github.com/user-attachments/assets/0fab7ede-7f9c-4903-93a1-3a9e96177c55)

teolemon commented:
Yes, that was in the case we had bulk products on the receipt. 
We've also received email reports on online proofs https://prices.openfoodfacts.org/prices/80341
![image](https://github.com/user-attachments/assets/c7be93fe-b4a9-4965-9fc4-b273c3e17f18)
```
Raphael Odini
  Friday at 12:47
could be because the mobile app doesn't manage prices coming from online stores (the case for this price)
could be because the mobile app doesn't manage the new discount_type field that was added this week (but this price doesn't have a discount, so there would be a bug on all other prices ?)
```

monsieurtanuki commented:
> Yes, that was in the case we had bulk products on the receipt.

I'm probably missing something, but I cannot see the negative impact of products added in bulk, for the current PR.

> We've also received email reports on online proofs

I've just created an issue in off-dart - curiously the AI didn't.

@teolemon Back to the current PR: what are the needed changes?

monsieurtanuki commented:
Thank you @teolemon for your review and feedbacks!
